### PR TITLE
feat: add OS version major/minor integer constants

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28061

**Summary:**
Added new integer constants...
- `Ti.Platform.versionMajor`
- `Ti.Platform.versionMinor`
- `OS_VERSION_MAJOR`
- `OS_VERSION_MINOR`

**Reason:**
Makes it easier to implement new OS features. For example...
```javascript
if (OS_IOS && (OS_VERSION_MAJOR >= 13)) {
	// Do something on iOS 13 or higher only.
}
```

---
**Test:**
_Run this test on different Android/iOS versions._
1. Build and run the below on Android or iOS.
2. Verify `versionMajor` and `versionMinor` match `Ti.Platform.version` string.
3. Verify `OS_VERSION_MAJOR` and `OS_VERSION_MINOR` match `Ti.Platform.version` string.

```javascript
var window = Ti.UI.createWindow({ layout: "vertical" });
window.add(Ti.UI.createView({ height: OS_IOS ? "20dp" : "10dp" }));
window.add(Ti.UI.createLabel({
	text: "Ti.Platform.version: " + Ti.Platform.version,
	top: "10dp",
}));
window.add(Ti.UI.createLabel({
	text: "Ti.Platform.versionMajor: " + Ti.Platform.versionMajor,
	top: "10dp",
}));
window.add(Ti.UI.createLabel({
	text: "Ti.Platform.versionMinor: " + Ti.Platform.versionMinor,
	top: "10dp",
}));
window.add(Ti.UI.createLabel({
	text: "OS_VERSION_MAJOR: " + OS_VERSION_MAJOR,
	top: "10dp",
}));
window.add(Ti.UI.createLabel({
	text: "OS_VERSION_MINOR: " + OS_VERSION_MINOR,
	top: "10dp",
}));
window.open();
```

### CODE REVIEW

File: apidoc/Titanium/Platform/Platform.yml [hunk 1]
```diff
@@ -357,10 +357,23 @@ properties:
...
+    type: Number
+    permission: read-only
+    since: "9.2.0"
+
+  - name: versionMinor
+    summary: The operating system's minor version number.
+    description: Will return zero if OS does not have a minor version.
+    type: Number
+    permission: read-only
+    since: "9.2.0"
```
Comment (sgtcoolguy) [inline apidoc/Titanium/Platform/Platform.yml:375]:
``` yml
default: `0`
```

Note that we haven't really tightened the definition for what is a valid "default" value. So it's really just for doc consumption on the site - but ideally we'd be able to more tightly specify this in some way that we could auto-generate tests or something from it.

File: iphone/Resources/app.js [hunk 2]
```diff
@@ -20,3 +20,6 @@ btn.addEventListener('click', function() {
 
 win.add(btn);
 win.open();
+
+console.log("@@@ OS_VERSION_MAJOR: " + OS_VERSION_MAJOR);
```
Comment (sgtcoolguy) [inline iphone/Resources/app.js:24]: Need to revert these changes

File: apidoc/Global/Global.yml [hunk 3]
```diff
@@ -320,6 +320,25 @@ properties:
...
+            // Do something on iOS 13 or higher only.
+        }
+        ```
+    type: Number
+    permission: read-only
+    since: "9.2.0"
+
+  - name: OS_VERSION_MINOR
+    summary: The operating system's minor version number.
+    description: Will return zero if OS does not have a minor version.
```
Comment (sgtcoolguy) [inline apidoc/Global/Global.yml:337]: link/refer to `Ti.Platform.versionMinor`?

File: apidoc/Global/Global.yml [hunk 4]
```diff
@@ -320,6 +320,25 @@ properties:
     permission: read-only
     since: "9.0.0"
 
+  - name: OS_VERSION_MAJOR
+    summary: The operation system's major version number.
```
Comment (sgtcoolguy) [inline apidoc/Global/Global.yml:324]: link/refer to `Ti.Platform.versionMajor`?

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]
This is awesome! It really shows how we're making it easier to add features cross-platform more easily too - the global defines in JS, the simplified JSC API on iOS, the already simple way to declare the property on Android.

My feedback is all nitpicks and none of it is required **except** the rollback of the change to the iOS app.js used by Xcode.

**Comment by jquick-axway** [comment]
Updated PR:
- Updated API docs.
- Updated unit tests to verify `versionMajor`/`versionMinor` match components in `version` string.
- Removed test log message from iOS test app. _(My bad.)_

File: apidoc/Titanium/Platform/Platform.yml (hunk 1 above)
Comment (jquick-axway) [reply apidoc/Titanium/Platform/Platform.yml:381]: Good idea. I'll do the same for `OS_VERSION_MINOR`.

**Comment by jquick-axway** [comment]
@sgtcoolguy, what do you think of us adding our `OS_ANDROID` and `OS_IOS` constants to the new "globals.js" file?
```javascript
Object.defineProperties(global, {
	'OS_ANDROID': { value: OS_ANDROID, writable: false },
	'OS_IOS': { value: OS_IOS, writable: false },
	OS_VERSION_MAJOR: { value: Ti.Platform.versionMajor, writable: false },
	OS_VERSION_MINOR: { value: Ti.Platform.versionMinor, writable: false }
});
```

The reason to do this is because these constants are **undefined** if we're **NOT** running babel on the app project. This can happen:
- In our `titanium_mobile/android/app` test project under Android Studio.
- In our `titanium_mobile/iphone/Resources` test project under Xcode.
- In a Titanium app project if "tiapp.xml" has `<transpile>false</transplie>` set.

**Comment by sgtcoolguy** [comment]
> @sgtcoolguy, what do you think of us adding our `OS_ANDROID` and `OS_IOS` constants to the new "globals.js" file?
> 
> ```js
> Object.defineProperties(global, {
> 	'OS_ANDROID': { value: OS_ANDROID, writable: false },
> 	'OS_IOS': { value: OS_IOS, writable: false },
> 	OS_VERSION_MAJOR: { value: Ti.Platform.versionMajor, writable: false },
> 	OS_VERSION_MINOR: { value: Ti.Platform.versionMinor, writable: false }
> });
> ```

Yeah, I'm totally fine with that. Makes sense to me.

**Review by sgtcoolguy — APPROVED** [review]

**Comment by sgtcoolguy** [comment]
This feels very much like a PR that we should be able to merge without a formal QE FR, since we have unit tests to cover usage.

**Comment by jquick-axway** [comment]
@sgtcoolguy , I agree.
